### ISSUE TITLE
enforced version upgrades should be relaxed to 2 minor versions at a time

### ISSUE THREAD

**Issue by slimsag** [sourcegraph/sourcegraph#9056]
Currently we enforce 1 minor version upgrade at a time. I fear this is going to come back to bite us in a few months when we find a user on e.g. 3.12 has not upgraded to 3.18 and must perform 6 different upgrades.

I think it is within our capability to support 2 minor versions (we did in the past without many issues) and that reduces this upgrade friction by 2x, so it seems worth it to me.

TODOs if we adopt this:

- Change documentation to 2 minor versions instead of 1.
- Change upgrade enforcement code to 2 minor versions instead of 1.

**Comment by unknwon** [comment]
cc @beyang

**Comment by keegancsmith** [comment]
I agree, 2 versions is easy to support. Just means you deprecate things a bit slower.

**Comment by slimsag** [comment]
@pecigonzalo I think we should do this in 3.19

**Comment by bobheadxi** [comment]
Related: https://github.com/sourcegraph/sourcegraph/pull/14974
